kafka-connect-s3/src/main/java/io/confluent/connect/s3/TopicPartitionWriter.java

```diff
@@ -352,7 +352,7 @@ private boolean checkRotationOrAppend(
   }
 
   private void commitOnTimeIfNoData(long now) {
-    if (buffer.isEmpty()) {
+    while (buffer.isEmpty()) {
```

Comment:
This function triggers only once when buffer becomes empty and then the connector enters into an idle state until a new record consumed. By changing it to `while`, I force the connector to keep commit on flush interval even if no data arrived, and by that make sure to auto resolve issues with former kafka commit failures.